sentiment/reddit_api.py

```diff
@@ -572,74 +557,69 @@ def wsb_community(l_args):
-                    s_datetime = datetime.utcfromtimestamp(submission.created_utc).strftime("%d/%m/%Y %H:%M:%S")
-                    s_link = f"https://www.reddit.com{submission.permalink}"
-                    s_all_awards = ""
-                    for award in submission.all_awardings:
-                        s_all_awards += f"{award['count']} {award['name']}\n"
-                    s_all_awards = s_all_awards[:-2]
-
-                    # Create dictionary with data to construct dataframe allows to save data
-                    d_submission[submission.id] = {
-                                                    'created_utc': s_datetime,
-                                                    'subreddit': submission.subreddit,
-                                                    'link_flair_text': submission.link_flair_text,
-                                                    'title':submission.title,
-                                                    'score': submission.score,
-                                                    'link': s_link,
-                                                    'num_comments': submission.num_comments,
-                                                    'upvote_ratio': submission.upvote_ratio,
-                                                    'awards': s_all_awards
-                                                }
-
-                    # Print post data collected so far
-                    print(f"{s_datetime} - {submission.title}")
-                    print(f"{s_link}")
-                    t_post = PrettyTable(['Subreddit', 'Flair', 'Score', '# Comments', 'Upvote %', "Awards"])
-                    t_post.add_row([submission.subreddit, submission.link_flair_text, submission.score,
-                                    submission.num_comments,f"{round(100*submission.upvote_ratio)}%", s_all_awards])
-                    print(t_post)
-                    print("")
-            # Check if search_submissions didn't get anymore posts
-            else:
-                break
+    (ns_parser, l_unknown_args) = parser.parse_known_args(l_args)
+
+    if l_unknown_args:
+        print(f"The following args couldn't be interpreted: {l_unknown_args}\n")
+        return
+
+    praw_api = praw.Reddit(client_id=cfg.API_REDDIT_CLIENT_ID,
+                            client_secret=cfg.API_REDDIT_CLIENT_SECRET,
+                            username=cfg.API_REDDIT_USERNAME,
+                            user_agent=cfg.API_REDDIT_USER_AGENT,
+                            password=cfg.API_REDDIT_PASSWORD)
+
+    d_submission = {}
+    l_watchlist_links = list()
+
+    psaw_api = PushshiftAPI()
+
+    if ns_parser.b_new:
+        submissions = praw_api.subreddit('wallstreetbets').new(limit=ns_parser.n_limit)
+    else:
+        submissions = praw_api.subreddit('wallstreetbets').hot(limit=ns_parser.n_limit)
+
+    while True:
+        submission = next(submissions, None)
+        if not submission:
```

Comment:
Out of curiosity, is this just style preference? Or there's a reason why this is better?

Also, I think you forgot to update the other functions on this file to be coherent with this

Reply:
Style preference – an early `continue` avoids nesting most of the block. I'll look into updating the others in the ULTRA MODE PR, since I think it deduplicates this code among others.